playbooks/roles/filebeat/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+
+- name: Install apt package dependencies.
+  apt:
+    name: apt-transport-https
+    state: present
+
+- name: Add Elastic's apt key
+  apt_key:
+    url: https://artifacts.elastic.co/GPG-KEY-elasticsearch
+    state: present
+
+- name: Add Elastic's apt repository
+  apt_repository:
+    repo: 'deb https://artifacts.elastic.co/packages/6.x/apt stable main'
+    state: present
+    update_cache: yes
+
+- name: Install Filebeat
+  apt:
+    name: filebeat
+    state: present
+
+- name: Install Filebeat configuration
+  template:
+    src: filebeat.yml.j2
+    dest: "{{ filebeat_config_dir }}/filebeat.yml"
+  notify:
+    - restart filebeat
+
+- name: Install Filebeat CA certificate
+  copy:
+    content: "{{ filebeat_ca_cert }}"
+    dest: "{{ filebeat_ca_path }}"
+    owner: root
+    group: root
+    mode: 0400
+  when: filebeat_ca_cert != "" and filebeat_ca_cert_url == ""
+
+- name: Install Filebeat CA certificate from URL
```

Comment:
There are some cases where adding the public cert through a var is pretty ugly or I think easy to mess up, like for Ocim appservers. I don't want to have to squeeze that public cert inside Ocim's `.env` somehow so that it can propgate to appservers running the playbook.

Reply:
An alternative would be to configure the URL in Ocim and make Ocim download the cert and render it in the settings, but the solution you have is fine, too.